lib/commands/submit_order/params_from_form.js

```diff
@@ -0,0 +1,133 @@
+'use strict'
+
+const colors = require('colors')
+const _isEmpty = require('lodash/isEmpty')
+const { Order } = require('bfx-api-node-models')
+const { Form, NumberPrompt } = require('enquirer')
+
+const orderTypeParam = require('../../params/order_type')
+const marketParam = require('../../params/market')
+const amountParam = require('../../params/amount')
+const priceParam = require('../../params/price')
+
+/**
+  * Renders a CLI form for supplying order parameters, and returns once
+  * sufficient parameters for the parsed type are supplied.
+  *
+  * @return {Object} params
+  */
+module.exports = async () => {
+  const params = {}
+  let parseComplete = false
+  let prompt
+
+  // get basic details
+  while (!parseComplete) {
+    prompt = new Form({
+      name: 'orderform',
+      message: 'Enter your order details:',
+      choices: [{
+        name: 'type',
+        message: 'Order Type',
+        initial: params.type || ''
+      }, {
+        name: 'symbol',
+        message: 'Symbol',
+        initial: params.symbol || ''
+      }, {
+        name: 'amount',
+        message: 'Amount',
+        initial: params.amount || ''
+      }]
+    })
+
+    const input = await prompt.run()
+
+    try {
+      params.type = orderTypeParam('order type', input.type)
+      params.symbol = marketParam('market', input.symbol)
+      params.amount = +amountParam('amount', _isEmpty(input.amount) ? NaN : +input.amount)
+      parseComplete = true
+    } catch (e) {
+      console.error(colors.red(e.message))
+      console.log('')
```

Comment:
empty log? just to add `\n`?

Reply:
Yeah to make the output easier to read; `bitfinex-api-node` uses `debug` by default which emits a lot of noise and is not meant for data display, we override it here: https://github.com/f3rno/bfx-cli/blob/master/lib/override_bfx_api_node_debug.js

For UI we have console + `colors` module, `enquirer` for input, and `blessed-contrib` for things like OB display. Overall we need to improve the UI but things like extra newlines make a difference (80/20)